Match: support for supplying multiple app identifiers at once

jverdi opened the pull request:
🔑 Allows you to supply multiple app identifiers to match at once and it will only perform (1) clone of your repo, and (1) check for your associated certificate.

As an action:

```
match(type: "appstore", 
      app_identifier: ["tools.fastlane.app", "tools.fastlane.app2", "tools.fastlane.app3"])
```

I made sure to also continue supporting the old syntax, so this doesn't break existing configurations:

```
match(type: "appstore", app_identifier: "tools.fastlane.app")
```

Multiple apps via command-line:

```
match appstore -a tools.fastlane.app,tools.fastlane.app2,tools.fastlane.app3
```

jverdi committed:
Match: support for supplying multiple app identifiers at once

🔑 Allows you to supply multiple app identifiers to match at once and it will only perform (1) clone of your repo, and (1) check for your associated certificate

jverdi commented:
This addresses performance issues discussed in #4100

I don't see the need for a full matrix-like solution. Generally speaking, CI builds will be building for one environment at a time. The real pain point has been the proliferation of app ids for extensions.

KrauseFx committed:
Don’t hang on git clone command when user credentials are missing (#6068)

Using the GIT_TERMINAL_PROMPT env variable the command will fail instead of prompting for the user’s credentials, it’s like magic

KrauseFx commented on match/README.md:
````diff
@@ -174,20 +174,17 @@ match --help
 You can make this even easier using [fastlane](https://github.com/fastlane/fastlane/tree/master/fastlane) by creating a match lane like this:
 
 ```
 lane :match do
````
Ah, the lane shouldn't be called match actually, I submitted a separate PR to change that https://github.com/fastlane/fastlane/pull/6074, make sure to rebase after it got merged

KrauseFx commented on match/lib/match/options.rb:
```diff
@@ -31,7 +31,8 @@ def self.available_options
                                      env_name: "MATCH_APP_IDENTIFIER",
-                                     description: "The bundle identifier of your app",
+                                     description: "The bundle identifier(s) of your app (comma-separated)",
+                                     is_string: false,
```
You should be able to use
```
type: Array
```

KrauseFx replied:
Oh, never mind, the user can pass either an array or String 👍 This is ok then

KrauseFx commented on match/lib/match/runner.rb:
```diff
@@ -10,25 +10,45 @@ def run(params)
-      spaceship.bundle_identifier_exists(params) if spaceship
+      app_identifiers.each do |app_identifier|
+        params[:app_identifier] = app_identifier
+        spaceship.bundle_identifier_exists(params) if spaceship
```
We could just skip the whole looping if no spaceship is defined, right? 👍

KrauseFx commented:
I love this approach: very little change, and very effective. I wonder if there are any other side effects we should think about?
I added some comments before it's ready to be merged 👍

kndl committed:
Add project_directory option to carthage action (#6072)

* Add project_directory option to carthage action

Accept an option of project_directory that adds a `--project-directory` flag to
the executed carthage command.

* Remove trailing whitespace.

johnboiles committed:
Escape `.` in `KEY` for use in `plutil` (#6067)

This is because plutil interprets dots as a separator for nested keys. So literal `.`s need to be escaped.

KrauseFx committed:
[match] Rename match lane (#6074)

Since we now show a warning if a lane is called after a tool

jverdi commented:
Re-opened with feedback incorporated as: https://github.com/fastlane/fastlane/pull/6080
